src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,48 +1,92 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { getTodos, getUser } from '../../api';
+import { User } from '../../types/User';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  userId: number;
+  todoId: number;
+  fnSelectTodo: (selectTodoId: number) => void;
+};
+
+export const TodoModal: React.FC<Props> = (
+  {
+    userId,
+    todoId,
+    fnSelectTodo,
+  },
+) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+
+  const loadData = async () => {
+    await Promise.all([
+      getUser(userId).then(currentUser => setUser(currentUser)),
+      getTodos().then(myTodos => setTodos(myTodos)),
+    ]);
+  };
+
+  useEffect(() => {
+    loadData();
+  }, []);
+
+  const filteredTodos = todos.filter(todo => todo.id === todoId);
 
-export const TodoModal: React.FC = () => {
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      {true ? (
-        <Loader />
-      ) : (
+      {user ? (
         <div className="modal-card">
-          <header className="modal-card-head">
-            <div
-              className="modal-card-title has-text-weight-medium"
-              data-cy="modal-header"
-            >
-              Todo #2
-            </div>
-
-            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-            <button
-              type="button"
-              className="delete"
-              data-cy="modal-close"
-            />
-          </header>
-
-          <div className="modal-card-body">
-            <p className="block" data-cy="modal-title">
-              quis ut nam facilis et officia qui
-            </p>
-
-            <p className="block" data-cy="modal-user">
-              {/* <strong className="has-text-success">Done</strong> */}
-              <strong className="has-text-danger">Planned</strong>
-
-              {' by '}
-
-              <a href="mailto:Sincere@april.biz">
-                Leanne Graham
-              </a>
-            </p>
-          </div>
+          {filteredTodos.map(({
```

Comment:
Do you really need to map todos in the TodoModal component?
Don't forget about key attribute too